app/prepare_proposal.go

```diff
@@ -32,7 +32,23 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		ante.DefaultSigVerificationGasConsumer,
 		app.IBCKeeper,
 	)
-	txs := FilterTxs(sdkCtx, handler, app.txConfig, req.BlockData.Txs)
+
+	var txs [][]byte
+	if app.LastBlockHeight() == 0 {
+		// the first block is expected to be empty; hence return an empty set of txs
+		txs = make([][]byte, 0)
+		if len(req.BlockData.Txs) != 0 {
+			// if the consensus layer sends non-empty set of transactions for
+			// block height 1, log it
+			app.Logger().Info(
+				"non-empty txs received from the consensus layer for block height 1",
+				"invalid",
+				len(req.BlockData.Txs),
```

Comment:
```suggestion
				"non-empty txs received from the consensus layer for block height 1",
				"numberOfTransactions",
				len(req.BlockData.Txs),
```

Reply:
Thanks, fixed it in https://github.com/celestiaorg/celestia-app/pull/2233/commits/93692c5d4876e18d3c4f366e8d9126df608679c4.